docs/reference/mapping.asciidoc

```diff
@@ -96,7 +96,14 @@ causing a mapping explosion:
     all nested fields, defaults to 10000. Indexing one document with an array of
     100 objects within a nested field, will actually create 101 documents, as
     each nested object will be indexed as a separate hidden document.
-
+`index.mapping.field.name.length.limit`::
+    Setting for the maximum length of a field name. The default value is
+    Long.MAX_VALUE (no limit).
+    *Note*
```

Comment:
Thanks for moving this, small follow up nit-pick: I don't really like adding the two points mentioned here as special "Note" with sub-bulltest here beacuse I think it draws too much attention to an otherwise quite marginal setting. I think it should be enough to either just inline the two additional sentences into the previous paragraph and/or put the first sentence in front of the setting description.